src/allocation_engine/interval_tree.rs

```diff
@@ -37,12 +40,93 @@ pub struct Constraint {
 /// memory slot that will be assigned to a device by the VMM.
 /// This structure represents the key of the Node object in
 /// the interval tree implementation.
-#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Hash, Ord)]
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Hash, Ord, Debug)]
 pub struct Range {
     pub min: u64,
     pub max: u64,
 }
 
+impl Range {
+    /// Create a new Range object.
+    pub fn new(min: u64, max: u64) -> Result<Self> {
```

Comment:
A `range` is an interval, so it's better defined with `start` and `end` instead of min/max. It also makes the code easier to read later on.

If we have a constructor, we should make `min` and `max` fields private (and rename them to start/end as well).